src/flight_paths/flight_paths.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+import math
+import requests
+
+
+FLIGHTS_URL = "https://codefellows.github.io/sea-python-401d4/_downloads/cities_with_airports.json"
+
+
+def calculate_distance(point1, point2):
+    """
+    Calculate the distance (in miles) between point1 and point2.
+    point1 and point2 must have the format [latitude, longitude].
+    The return value is a float.
+
+    Modified and converted to Python from: http://www.movable-type.co.uk/scripts/latlong.html
+    """
+
+    def convert_to_radians(degrees):
+        """Convert lat lon to radians."""
+        return degrees * math.pi / 180
+
+    radius_earth = 6.371E3  # km
+    phi1 = convert_to_radians(point1[0])
+    phi2 = convert_to_radians(point2[0])
+
+    delta_phi = convert_to_radians(point1[0] - point2[0])
+    delta_lam = convert_to_radians(point1[1] - point2[1])
+
+    a = math.sin(0.5 * delta_phi)**2 + math.cos(phi1) * math.cos(phi2) * math.sin(0.5 * delta_lam)**2
+    c = 2 * math.atan2(math.sqrt(a), math.sqrt(1 - a))
+    return radius_earth * c / 1.60934  # convert km to miles
+
+
+def get_flight_json():
+    """Return json of flight info."""
+    flight_json = requests.get(FLIGHTS_URL).json()
+    return flight_json
+
+
+def create_flight_graph(flight_json):
+    """Return dictionary with city as key and values are connecting cities and lat lon."""
+    flight_dict = {}
+    lat_lon_dict = {}
+    for idx in range(len(flight_json)):
+        key = flight_json[idx]['city']
+        value = flight_json[idx]['destination_cities']
+        value2 = flight_json[idx]['lat_lon']
+        flight_dict[key] = value
+        lat_lon_dict[key] = value2
+    return flight_dict, lat_lon_dict
+
+
+def find_path(flight_dict, start_city, end_city, path=[]):
+    """Return a flight path between two cities."""
+    stack = [start_city]
+    while stack:
+        cur_city = stack.pop()
+        if cur_city not in flight_dict:
+            return None
+        if cur_city not in path:
+            path.append(cur_city)
+            for city in flight_dict[cur_city]:
+                if city == end_city:
+                    path.append(end_city)
+                    return path
+            for city in flight_dict[cur_city]:
```

Comment:
You know that you don't need to do this loop twice right? It's computationally expensive. ON THE OTHER HAND! It makes for a better output when looking for a path between two cities. Don't really care about the cities that you don't need to visit if the city you want is an immediate neighbor right?